java/src/com/mongodb/csfle/CSFLEHelpers.java

```diff
@@ -0,0 +1,219 @@
+                        .append("subType", "04")));
+        return new Document().append("keyId", keyIds);
+    }
+
+    public static Document createJSONSchema(String keyId) throws IllegalArgumentException {
+        if (keyId.isEmpty()) {
+            throw new IllegalArgumentException("keyId must contain your base64 encryption key id.");
+        }
+        return new Document().append("bsonType", "object").append("encryptMetadata", createEncryptMetadataSchema(keyId))
+                .append("properties", new Document()
+                        .append("ssn", buildEncryptedField(keyId, "int", true))
+                        .append("bloodType", buildEncryptedField(keyId, "string", false))
+                        .append("medicalRecords", buildEncryptedField(keyId, "array", false))
+                        .append("insurance", new Document()
+                                .append("bsonType", "object")
+                                .append("properties",
+                                        new Document().append("policyNumber", buildEncryptedField(keyId, "int", true)))));
+    }
+
+    // Creates Normal Client
+    public static MongoClient createMongoClient(String connectionString) {
+        return MongoClients.create(connectionString);
+    }
+
+    // Creates KeyVault which allows you to create a key as well as encrypt and decrypt fields
+    public static ClientEncryption createKeyVault(String connectionString, String kmsProvider, byte[] localMasterKey, String keyVaultCollection) {
+        Map<String, Object> masterKeyMap = new HashMap<String, Object>();
+        masterKeyMap.put("key", localMasterKey);
+        Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>();
+        kmsProviders.put(kmsProvider, masterKeyMap);
+
+        ClientEncryptionSettings clientEncryptionSettings = ClientEncryptionSettings.builder()
+                .keyVaultMongoClientSettings(MongoClientSettings.builder()
+                        .applyConnectionString(new ConnectionString(connectionString))
+                        .build())
+                .keyVaultNamespace(keyVaultCollection)
+                .kmsProviders(kmsProviders)
+                .build();
+
+        return ClientEncryptions.create(clientEncryptionSettings);
+    }
+
+    // You may need to update the following variable to point to your mongocryptd binary
+    private static String mongocryptdPath = "/usr/local/bin/mongocryptd";
+
+    // Creates Encrypted Client which performs automatic encryption and decryption of fields
+    public static MongoClient createEncryptedClient(String connectionString, String kmsProvider, byte[] masterKey, String keyVaultCollection, Document schema, String dataDb, String dataColl) {
+        String recordsNamespace = dataDb + "." + dataColl;
+
+        Map<String, BsonDocument> schemaMap = new HashMap<String, BsonDocument>();
+        schemaMap.put(recordsNamespace, BsonDocument.parse(schema.toJson()));
```

Comment:
This `BsonDocument.parse` is weird.  Either pass in a JSON string to this method, or pass in the a BsonDocument instead of a Document.

Reply:
Ping on this.

Reply:
Sorry, missed this. I'm not sure how to implement the suggestion. `AutoEncryptionSettings.schemaMap` requires a `Map<String, BsonDocument>`. Is there a better way to convert the `Document` to `BsonDocument`? Or is this related to this comment? https://github.com/mongodb-university/csfle-guides/pull/2#discussion_r349787332
If the latter, I can open another JIRA ticket to make that update since it requires several updates to the guide.

Reply:
You could have `createJSONSchema` return a BsonDocument instead of a Document.  It's a bit more cumbersome to construct BsonDocument instances, but shouldn't be too different than Document.